app.py

```diff
@@ -0,0 +1,118 @@
+import os
+from datetime import datetime, timedelta
+
+import jwt
+from dotenv import load_dotenv
+from flask import Flask, redirect, request, session, url_for
+
+load_dotenv()
+
+
+# Custom exceptions
+class JWTError(Exception):
+    """Base class for JWT related exceptions."""
+
+
+class JWTGenerationError(JWTError):
+    """Exception raised for errors during JWT generation."""
+
+
+class JWTDecodingError(JWTError):
+    """Exception raised for errors during JWT decoding."""
+
+
+# Constants
+SECRET_KEY = os.getenv("FASTAPI_EXAMPLE_SECRET_KEY")
+APP_ID = os.getenv("FASTAPI_EXAMPLE_APP_ID", "flask_example")
+ALGORITHM = os.getenv("FASTAPI_EXAMPLE_ALGORITHM", "RS256")
+SERVER_PUBLIC_KEY_FILE = os.getenv(
+    "FASTAPI_EXAMPLE_SERVER_PUBLIC_KEY_FILE", "keys/server_public_key.pem"
+)
+CLIENT_PRIVATE_KEY_FILE = os.getenv(
+    "FASTAPI_EXAMPLE_CLIENT_PRIVATE_KEY_FILE", "keys/client_private_key.pem"
+)
+SESSION_LIFETIME = int(os.getenv("FASTAPI_EXAMPLE_SESSION_LIFETIME", 300))
+TOKEN_EXPIRES_IN_SECONDS = int(
+    os.getenv("FASTAPI_EXAMPLE_TOKEN_EXPIRES_IN_SECONDS", 300)
+)
+MEMBRANE_SERVER = os.getenv("FASTAPI_EXAMPLE_MEMBRANE_SERVER", "http://127.0.0.1:5000")
```

Comment:
supposed to be optional but doesn't look like your code works without it?

Reply:
Done